### ISSUE TITLE
Revising minimal build for CI

### ISSUE THREAD

**Issue by matthewfeickert** [lwtnn/lwtnn#130]
@dguest As the "minimal" CI builds are now failing

```log
-- Build files have been written to: /home/runner/work/lwtnn/lwtnn/build
[  1%] Creating directories for 'Eigen'
[  3%] Creating directories for 'Boost'
[  5%] Performing download step (download, verify and extract) for 'Eigen'
-- Downloading...
   dst='/home/runner/work/lwtnn/lwtnn/build/externals/src/eigen-3.3.7.tar.bz2'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.bz2'
[  7%] Performing download step (download, verify and extract) for 'Boost'
-- Downloading...
   dst='/home/runner/work/lwtnn/lwtnn/build/externals/src/boost_1_64_0.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.gz'
-- verifying file...
       file='/home/runner/work/lwtnn/lwtnn/build/externals/src/eigen-3.3.7.tar.bz2'
-- Downloading... done
-- extracting...
     src='/home/runner/work/lwtnn/lwtnn/build/externals/src/eigen-3.3.7.tar.bz2'
     dst='/home/runner/work/lwtnn/lwtnn/build/externals/src/Eigen'
-- extracting... [tar xfz]
-- [download 0% complete]
CMake Error at Boost-stamp/download-Boost.cmake:170 (message):
  Each download failed!
```

it seems that the idea of the "minimal" build only comes up in 

https://github.com/lwtnn/lwtnn/blob/8f1e9a3084f26605714c8ee162fcf74daf0a0dc8/tests/build-cmake.sh#L27-L29

then the idea of running a "minimal" build in a CI job seems a bit weird as that is making assumptions on what the image should already contain, no? Wouldn't it be better to just drop this test from CI as it makes assumptions on installed software?

**Comment by dguest** [comment]
I'm not sure I understand what makes this test weird. We have a [special preparation for the "minimal" build](https://github.com/lwtnn/lwtnn/blob/master/.github/workflows/ci.yml#L51) that doesn't download eigen and boost to see if this works. The name is a bit weird, because actually the minimal build does a lot more, but it's supposed to require less software on the system.

**Comment by matthewfeickert** [comment]
> The name is a bit weird, because actually the minimal build does a lot more, but it's supposed to require less software on the system (it gets and builds a local version from via CMake).

Oh. I misunderstood. "minimal" means "minimal pre-installed software required" not "minimal dependencies required".

**Comment by dguest** [comment]
No, but it does mean that we have a dependency outside github somewhere that we don't pull from in the other jobs. It's possible that this is just a temporary glitch so I'm retrying the job. The boost version is also pinned, so it's pretty weird to have it suddenly fail.

**Comment by dguest** [comment]
The error is
```
    error: downloading 'https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.tar.gz' failed
          status_code: 22
          status_string: "HTTP response code said error"
```
I'm not too familear with bintray... and I'm not sure why we're using this rather than https://boostorg.jfrog.io/artifactory/main/release/1.64.0/source/boost_1_64_0.tar.gz like it says on the boost website.

**Comment by matthewfeickert** [comment]
> I'm not too familear with bintray... and I'm not sure why we're using this rather than https://boostorg.jfrog.io/artifactory/main/release/1.64.0/source/boost_1_64_0.tar.gz like it says on the boost website.

Cool. jfrog is decently new, so maybe this is a new change in the directions? Hopefully this is a 1liner PR to fix. :)

**Comment by dguest** [comment]
See #131

**Comment by matthewfeickert** [comment]
Resolved by PR #131
